### ISSUE TITLE
apc.php diagrams broken 

### ISSUE THREAD

**Issue by teo1978** [krakjoe/apcu#140]
![screenshot from 2015-11-02 22 47 35](https://cloud.githubusercontent.com/assets/542107/10895307/c555bd24-81b3-11e5-9b62-195ac7e5e832.png)

The diagrams showing free and used memory in apc.php are broken. In the pie chart, there are empty zones, and in the fragmentation chart there are black blocks. All should be either green (free) or orange (used). Zones that are neither used nor free don't make sense, or if they do, some indication should be given of what they are supposed to mean.

Example screenshot attached.

**Comment by krakjoe** [comment]
Could you try with latest pecl release ?

**Comment by krakjoe** [comment]
Bump, can you try with 4.0.10 or 5.1.2 ?

**Comment by mustangxu** [comment]
Same issue with 5.1.3

**Comment by krakjoe** [comment]
I don't think there is any bug here.

What is happening is that the memory is fragmented, this happens when lots of alloc/free happens and the allocator has lots of little small chunks of memory that are not large enough to satisfy any allocation requests.

I'm going to close the bug because it seems like normal behaviour to me, and no evidence is forthcoming to the contrary.

If you believe you have an actual bug, please open a new bug report.

**Comment by staabm** [comment]
> Zones that are neither used nor free don't make sense, or if they do, some indication should be given of what they are supposed to mean.

I feel this is a valid point.

**Comment by teo1978** [comment]
There IS a bug, or at least a terrible design here.
The pie chart should have a third color (e.g. gray) for "Wasted", and all the parts that aren't either used or free should be of that color.

The "Detailed usage and fragmentation" diagram in my example is very confusing.
(1) What is the black supposed to mean? Are those thousands of orange or green 1-px-height rectangles with 1px black border, or are they actual black areas representing wasted (unused, non-available) memory? If it is the former, then the border should be eliminated; if it is the latter, then again, a more intuitive color such as grey should be used with the corresponding legend.
(3) Why aren't the visible green fragment connected via a line to a number indicating the size like the small invisible fragments at the beginning? I assume it is because you only show up details up to a certain maximum number of fragments, to avoid a potentially infinite size image. If so, then (a) there should be an indication of that, such as an ellipsis sign ("...") or something like "N more fragments", (b) instead of showing the first segments, you should pick the biggest ones.

**Comment by krakjoe** [comment]
Remember that we inherited this code from APC, that it's very old, not very nice to work on, and been downloaded a million billion times or whatever.

PR's are very welcome. 

When I took on APCu, many people said they were going to do a status page for opcache+apcu, because no such status page existed for opcache at the time. There are several projects that provide a status page for opcache now, but apc has been left behind.

Anyone who works closely with me will tell you that you don't want to use a graphical thing I made, this just isn't my thing.

Anyone is welcome to do better, I encourage you to try :)
